From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Thu, 25 Mar 2021 23:09:19 +0000
Subject: [PATCH] meson: Avoid a trailing slash on libdir and datadir

When package_subdir is empty. This gets passed into the produced
pkg-config file and then confuses meson.

https://gitlab.gnome.org/GNOME/gnome-characters/-/issues/86
---
 meson.build | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index 73c07a9..d06ed77 100644
--- a/meson.build
+++ b/meson.build
@@ -103,14 +103,18 @@ add_project_arguments(
 
 # Setup various paths that subdirectory meson.build files need
 package_subdir = get_option('package_subdir') # When used as subproject
-datadir        = get_option('datadir')  / package_subdir
-libdir         = get_option('libdir')  / package_subdir
-girdir         = get_option('datadir') / package_subdir / 'gir-1.0'
-typelibdir     = get_option('libdir')  / package_subdir / 'girepository-1.0'
 if package_subdir != ''
-  vapidir = get_option('datadir') / package_subdir / 'vapi'
+  datadir    = get_option('datadir') / package_subdir
+  libdir     = get_option('libdir')  / package_subdir
+  girdir     = get_option('datadir') / package_subdir / 'gir-1.0'
+  typelibdir = get_option('libdir')  / package_subdir / 'girepository-1.0'
+  vapidir    = get_option('datadir') / package_subdir / 'vapi'
 else
-  vapidir = get_option('datadir') / 'vala' / 'vapi'
+  datadir    = get_option('datadir')
+  libdir     = get_option('libdir')
+  girdir     = get_option('datadir') / 'gir-1.0'
+  typelibdir = get_option('libdir')  / 'girepository-1.0'
+  vapidir    = get_option('datadir') / 'vala' / 'vapi'
 endif
 
 glade_catalog_feature = get_option('glade_catalog')
